python_modules/libraries/dagster-components/dagster_components/core/defs_module.py

```diff
@@ -42,52 +44,89 @@ class ComponentFileModel(BaseModel):
 
 
 @dataclass
-class DefsModule(Component):
+class DefsModuleComponent(Component):
     path: Path
 
     @classmethod
-    def from_context(cls, context: ComponentLoadContext) -> Optional["DefsModule"]:
+    def from_context(cls, context: ComponentLoadContext) -> Optional["DefsModuleComponent"]:
         # this defines the priority of the module types
         module_types = (
             YamlComponentDefsModule,
             PythonicComponentDefsModule,
             DagsterDefsModule,
-            FolderDefsModule,
+            DefsFolderComponent,
         )
         module_filter = filter(None, (cls.from_context(context) for cls in module_types))
         return next(module_filter, None)
 
     @classmethod
-    def load(cls, attributes: Any, context: ComponentLoadContext) -> "DefsModule":
+    def load(cls, attributes: Any, context: ComponentLoadContext) -> "DefsModuleComponent":
         return check.not_none(cls.from_context(context))
 
 
 @dataclass
-class FolderDefsModule(DefsModule):
+class _DefsFolderComponentYaml(Resolvable):
+    asset_post_processors: Optional[Sequence[AssetPostProcessor]] = None
+
+
+@dataclass
+class DefsFolderComponent(DefsModuleComponent):
     """A folder containing multiple submodules."""
 
-    submodules: Sequence[DefsModule]
+    submodules: Sequence[DefsModuleComponent]
+    asset_post_processors: Optional[Sequence[AssetPostProcessor]]
 
     @classmethod
-    def from_context(cls, context: ComponentLoadContext) -> Optional["FolderDefsModule"]:
+    def get_schema(cls):
+        return _DefsFolderComponentYaml.model()
+
+    @classmethod
+    def load(cls, attributes: Any, context: ComponentLoadContext) -> "DefsFolderComponent":
+        # doing this funky thing because some of our attributes are resolved from the context,
+        # so we split up resolving the yaml-defined attributes and the context-defined attributes,
+        # meaning we manually invoke the resolution system here
+        resolved_attributes = _DefsFolderComponentYaml.resolve_from_model(
+            context.resolution_context.at_path("attributes"),
+            attributes,
+        )
+        return check.not_none(
+            DefsFolderComponent.from_context(
+                context, post_processors=resolved_attributes.asset_post_processors
+            )
```

Comment:
I'm not a huge fan of this chicancery I think we should do a different appraoch, even if it means having another wrapping class. Alternatively we can add a feature to `Resolved` to ignore certain fields

Reply:
Does this generate the correct schema for tooling etc? e.g. the `vscode` plugin?

Reply:
yep -- because I defined `get_schema()` and that's where all of our tooling gets its info from